### ISSUE TITLE
Minor version change (2.0.x -> 2.1.x) breaks API compatibility

### ISSUE THREAD

**Issue by soplakanets** [neumino/thinky#327]
Upgrading from 2.0.x to 2.1.x changed the way `.execute()` handles queries. At 2.0.13 the method returned cursor, while at 2.1.5 it automatically coerces results to array. 

In my opinion it a) should be reflected in documentation (currently it still shows examples with cursors) and b) breaking changes deserve major version bump since many people rely on semver(^2.0.0) in npm.

**Comment by soplakanets** [comment]
My friend just pointed me to this commit: https://github.com/neumino/thinky/commit/fabf210db4865d7bf3a53ca482ea55b881b5333c commit message is a bit misleading, so it might be a simple mistake.

**Comment by neumino** [comment]
Yea sorry, https://github.com/neumino/thinky/commit/fabf210db4865d7bf3a53ca482ea55b881b5333c is the culprit (and me).

I usually don't break changes between RethinkDB releases, meaning that 2.1.x is backward compatible with 2.1.x'. The only exception for bugs. On another note, you shouldn't expect compability between 2.x.0 and 2.x'.0. Thinky follows RethinkDB tagging and as far as I know they can still break compatibility between these versions.

About commit fabf210db4865d7bf3a53ca482ea55b881b5333c, I have to admit that I considered a bug a bit too fast. It was technically there for a long time (there was even tests for that) so whether it's actually a bug or not is debatable. I usually wait for 2.x releases to push that but didn't pay attention when I tried to fix #320.

**Comment by neumino** [comment]
TL;DR: Use ~2.1.0 and not ^2.1.0 in your package.json and you're usually good.
Except when I make a mistake -_-

**Comment by laggyluke** [comment]
Something to consider: npm uses `^2.1.5` notation automatically when you do `npm install --save thinky`.

There are actually two separate issues here:

1. A breaking change was pushed without bumping major version number - no big deal, we all make mistakes.
2. thinky follows RethinkDB tagging instead of semver - this is kinda unfortunate as semver is basically a must for npm packages.
